tycho-core/src/main/java/org/eclipse/tycho/p2tools/TychoMirrorApplication.java

```diff
@@ -103,17 +108,49 @@ protected boolean isApplicable(IInstallableUnit iu, IRequirement req) {
                         if (onlyFilteredRequirements && filter == null) {
                             return false;
                         }
-                        return !considerFilter || filter == null || filter.isMatch(selectionContext);
+                        return !considerFilter || filter == null || matchesSelectionContext(filter);
                     }
                 }
-                return super.isApplicable(req);
+                return permissiveSlicer_isApplicable(req);
+            }
+
+            private boolean permissiveSlicer_isApplicable(IRequirement req) {
```

Comment:
why not use the existing `isApplicable(IRequirement req)` and let it not throw USOE?

Reply:
Done.